packages/pyright-internal/src/analyzer/service.ts

```diff
@@ -917,9 +917,23 @@ export class AnalyzerService {
 
         timingStats.findFilesTime.timeOperation(() => {
             const matchedFiles = this._matchFiles(this._configOptions.include, this._configOptions.exclude);
+            if (matchedFiles.length == 0) {
+                const envPwd: string = (process.env.PWD as string);
```

Comment:
This is unsafe and will crash at runtime if `PWD` is unset. In TS, `as` is a cast with no type checks.

Reply:
Ah, thanks. unset PWD couldn't make it crash with vim, but maybe it's respawning it immediately. Anyway, I pushed the fix, hope it's correct.

Reply:
I'm afraid it's not; the correct fix would just be `const envPwd = process.env.PWD` and then `if (envPwd)` to check truthiness.

It's good if this works for you (and absolutely fork and use it), but I don't know that we're going to accept this PR or other changes to fix this quite yet until we spec out how we intend for the rootless scenario to work and compare it to our current behavior.

Reply:
Yes, sure. I'm not entirely sure it should be merged either. I was hoping I kick off something at least.
Thank you for the heads up and code explanation.